pkg/dao/rpms.go

```diff
@@ -452,3 +453,95 @@ func (r *rpmDaoImpl) ListSnapshotRpms(ctx context.Context, orgId string, snapsho
 	}
 	return response, total, nil
 }
+
+func (r *rpmDaoImpl) DetectRpms(orgID string, request api.DetectRpmsRequest) (*api.DetectRpmsResponse, error) {
+	if orgID == "" {
+		return nil, fmt.Errorf("orgID cannot be an empty string")
+	}
+	// verify length of URLs or UUIDs is greater than 1
+	if len(request.URLs) == 0 && len(request.UUIDs) == 0 {
+		return nil, fmt.Errorf("must contain at least 1 URL or 1 UUID")
+	}
+	// set limit if not already or if more than max
+	if request.Limit == nil {
+		request.Limit = pointy.Int(api.ContentUnitSearchRequestLimitDefault)
+	}
+	if *request.Limit > api.ContentUnitSearchRequestLimitMaximum {
+		request.Limit = pointy.Int(api.ContentUnitSearchRequestLimitMaximum)
+	}
+	uuids := request.UUIDs
+	var missingRpms []string
+	var dataResponse *api.DetectRpmsResponse
+	var detectRpmsModel *models.DetectRpmsResponse
+
+	// check that repository uuids exist
+	for _, uuid := range uuids {
+		found := models.RepositoryConfiguration{}
+		if err := r.db.
+			Where("uuid = ?", uuid).
+			First(&found).
+			Error; err != nil {
+			return dataResponse, &ce.DaoError{
+				BadValidation: true,
+				Message:       "Could not find repository with UUID: " + uuid,
+			}
+		}
+	}
+	// check that repository urls exist and handle tail chars
+	urls := make([]string, len(request.URLs)*2)
```

Comment:
what is the "*2" for here?

Reply:
ah this was when i using the logic from handleTailChars() in helpers.go before switching to using models.CleanupURL(). i forgot to take this out, thank you! removing